File: TUGAS 1 CPP/account.cpp
#include <iostream>
#include <iomanip>
using namespace std;

class Account {
    private :
        // deklarasi variabel privat
        int noRekening;
        double saldo;

    public :
        // konstruktor dengan nilai saldo default
        Account(int noRek) {
            this->noRekening = noRek;
            this->saldo = 0.0;
        }

        // konstruktor 
        Account (int noRek, double sld) {
            this->noRekening = noRek;
            this->saldo = sld;
        }

        // fungsi getter untuk mengembalikan nilai variabel noRekening privat
        int getNoRekening() const {
            return noRekening;
        }

        // fungsi getter untuk mengembalikan nilai variabel saldo privat
        double getSaldo() const {
            return saldo;
        }

        // fungsi setter untuk mengubah variabel saldo
        void setSaldo(double sld) {
            this->saldo = sld;
        }

        // fungsi untuk menambahkan jumlah pada saldo
        void kredit(double jumlah) {
            saldo += jumlah;
        }

        // fungsi untuk mengurangi jumlah pada saldo
        void debit(double jumlah) {
            if (jumlah <= saldo) {
                saldo -= jumlah;
            } else {
                cout << "Jumlah yang ditarik melebihi saldo saat ini!" << endl;
            }
        }

        // fungsi untuk menampilkan deskripsi Akun
        void cetakAccount() const {
            cout << fixed << setprecision(2);
            cout << "A/C no : " << noRekening << "  Saldo saat ini : $" << saldo << endl;
        }
};

int main() {
    // deklarasi konstruktor account ke-1
    Account account1(8111, 99.99);
    cout << "--- Account 1 ----" << endl;
    account1.cetakAccount(); // menampilkan output no rekening dan saldo pada account ke-1
    account1.kredit(20); // menambahkan jumlah saldo dengan menggunakan fungsi kredit
    account1.debit(10); // mengurangi jumlah saldo dengan menggunakan fungsi debit
    account1.cetakAccount(); // menampilkan output no rekening dan saldo terbaru

    // deklarasi konstruktor account ke-2
    Account account2(8222);
    cout << "\n--- Account 2 ----" << endl;
    account2.cetakAccount(); // menampilkan output no rekening dan saldo pada account ke-2 dengan nilai default
    account2.setSaldo(100); // mengubah/menambahkan nilai saldo dengan menggunakan fungsi setSaldo
    account2.kredit(20); // menambahkan jumlah saldo dengan menggunakan fungsi kredit
    account2.debit(200); // menampilkan pesan bahwa jumlah yg ditarik melebihi saldo terkini
    account2.cetakAccount(); // menampilkan output no rekening dan saldo terbaru

    return 0;
}

File: TUGAS 1 JAVA/account.java
public class account {
    // deklarasi variabel privat
    private int noRekening;
    private double saldo;

    // konstruktor account dengan noRekening yang diinputkan dan nilai saldo adalah 0
    public account(int noRekening) {
        this.noRekening = noRekening;
        this.saldo = 0.0;
    }

    // konstruktor account dengan noRekening dan saldo yang diinputkan
    public account(int noRekening, double saldo) {
        this.noRekening = noRekening;
        this.saldo = saldo;
    }

    // tidak menggunakan setter untuk noRekening dan saldo
    // noRekening tidak didesain untuk diubah nilainya
    // nilai saldo hanya dapat diubah melalui kredit dan debit
    // publik getter untuk variabel noRekening privat
    public int getNoRekening() {
        return this.noRekening;
    }
    
    // publik getter untuk variabel saldo privat
    public double getSaldo() {
        return this.saldo;
    }

    // menampilkan/mencetak nilai noRekening dan saldo
    public String toString() {
        return String.format("account[no. Rekening=%d, saldo=$%.2f]", noRekening, saldo);
    }

    // menghitung jumlah kredit dan menambah saldo
    public void kredit(double jumlah) {
        saldo += jumlah;
    }
    
    // menghitung jumlah debit dan mengurangi saldo jika saldo >= jumlah
    public void debit(double jumlah) {
        if (saldo >= jumlah) {
            saldo -= jumlah;
        } else {
            System.out.println("jumlah terlampaui");
        }
    }

    // mentransfer ke account lainnya jika saldo >= jumlah
    public void transferKe(double jumlah, account lainnya) {
        if (saldo >= jumlah) {
            this.saldo -= jumlah;
            lainnya.saldo += jumlah;
        } else {
            System.out.println("jumlah terlampaui");
        }
    }
}
